src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -192,7 +201,11 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         }
 
         if (labels.getOrNull()) {
-            buildImageCmd.withLabels(labels.get().collectEntries { [it.key, it.value.toString()] })
+            buildLabels.putAll(labels.get())
```

Comment:
So `labels` can silently override `buildLabels`? This may cause some nasty bugs I guess.

Reply:
No they will add to each other. They are complementary. If some labels are really part of the build image inputs, then they should be 'labels'. If they are just complementary informations, they should be 'buildLabels'